airbyte-bootloader/src/test/java/io/airbyte/bootloader/SecretMigratorTest.java

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.bootloader;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.collect.Lists;
+import io.airbyte.bootloader.SecretMigrator.ConnectorConfiguration;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.config.ConfigSchema;
+import io.airbyte.config.DestinationConnection;
+import io.airbyte.config.SourceConnection;
+import io.airbyte.config.StandardDestinationDefinition;
+import io.airbyte.config.StandardSourceDefinition;
+import io.airbyte.config.persistence.ConfigPersistence;
+import io.airbyte.config.persistence.split_secrets.SecretCoordinate;
+import io.airbyte.config.persistence.split_secrets.SecretPersistence;
+import io.airbyte.protocol.models.ConnectorSpecification;
+import io.airbyte.validation.json.JsonValidationException;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+public class SecretMigratorTest {
```

Comment:
It doesn't seem like any of these tests are fully testing a secret migration from beginning to end. Could we add a test here which takes in a real connector configuration with some secrets in it, and verifies that those secrets are properly migrated, without mocking out any of the functionality of the secret migrator?

Reply:
@benmoriceau bump on this - I think there would be value in having a test that doesn't mock out anything in the secret migrator (only mock out the SecretPersistence), and verifies that a real source/dest configuration with actual plaintext secrets get properly migrated